src/Containers/MiddleCanvas/Template.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { 
+    DragDropContext, 
+    Droppable,
+    DroppableProvided,
+    Draggable,
+    DraggableProvided,
+    DraggableStateSnapshot,
+    DropResult,
+} from 'react-beautiful-dnd';
+import { ITemplatePrefill } from '/MiddleCanvasTypes';
+import { TableComponent } from './TableComponent';
+import { DroppableElement } from './DroppableElement';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+
+export interface TemplateProps extends MainInterface, ResponsiveInterface {
+    isPreview?: boolean,
+    templatePrefills: ITemplatePrefill
+};
+
+export const Template: React.FC<TemplateProps> = ({
+    isPreview,
+    templatePrefills,
+    ...props
+}): React.ReactElement => {
+    const [items, setItems] = useState(Object.values(templatePrefills));
+
+    /**
+     * Reorders the draggable elements in a list
+     * @param {ITemplatePrefill} list - list of objects to reorder
+     * @param {number} startIndex - index of where the element originates from
+     * @param {number} endIndex - index of where the element will be placed
+     */
+    const reorder = (list: ITemplatePrefill, startIndex: number, endIndex: number): ITemplatePrefill => {
+        const result = Object.values(list);
+        const [removed] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removed);
+    
+        return result;
```

Comment:
Naming can be better here, result does not tell me much about what is happening.

Also 1, 0 are magic numbers.